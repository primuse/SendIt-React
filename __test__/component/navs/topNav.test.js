import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import TopNav from '../../../src/components/navs/topNav.jsx';

const initialState = {
  auth: {
    isAuthenticated: true,
    user: {
      firstName: 'Tiku',
      lastName: 'Okoye',
      isadmin: false
    },
  },
  users: {
    allUsers: [],
    show: false
  }
};
const mockStore = configureStore([thunk]);
const store = mockStore(initialState);

describe('Top Nav component', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<MemoryRouter>
        <Provider store={store}>
          <TopNav />
        </Provider>
      </MemoryRouter>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('Top nav component', () => {
  let topNav;
  beforeEach(() => {
    topNav = mount(<MemoryRouter>
      <Provider store={store}>
        <TopNav />
      </Provider>
    </MemoryRouter>);
  });
  afterEach(() => {
    topNav.unmount();
  });
  it('renders the Top Nav', () => {
    const toggle = topNav.find('#toggle');
    expect(toggle.length).toEqual(1);
    toggle.simulate('click');
  });
});
