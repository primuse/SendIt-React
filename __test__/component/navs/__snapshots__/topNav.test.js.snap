// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Top Nav component renders correctly 1`] = `
<nav
  id="nav"
>
  <a
    className="toggler mr-1"
    href="#"
    id="toggle"
    onClick={[Function]}
  >
    <img
      src="test-file-stub"
    />
  </a>
  <ul
    className="p-0"
  >
    <li>
      Dashboard
    </li>
  </ul>
  <ul
    className="p-0 ml-auto"
  >
    <li
      className="mr-1"
    >
      <a
        href="/profile"
        onClick={[Function]}
      >
        My Profile
      </a>
    </li>
    <li>
      <a
        href="/"
        onClick={[Function]}
      >
        Logout
      </a>
    </li>
  </ul>
</nav>
`;
